src/primitives/certificate.h

```diff
@@ -0,0 +1,237 @@
+#ifndef _CERTIFICATE_H
+#define _CERTIFICATE_H
+
+#include "transaction.h"
+#include "policy/fees.h"
+
+struct CMutableScCertificate;
+
+class CScCertificate : virtual public CTransactionBase
```

Comment:
Introduce an in memory only backwardTrasferStartIdx field and the isBackwardTransfer method